lib/kubeclient/kube_exception.rb

```diff
@@ -1,10 +1,11 @@
 # Kubernetes HTTP Exceptions
 class KubeException < StandardError
-  attr_reader :error_code, :message
+  attr_reader :error_code, :message, :response
 
-  def initialize(error_code, message)
+  def initialize(error_code, message, response = nil)
```

Comment:
Minor: I wouldn't set a default `nil` for `response`.
That means that you'd have to explicitly set it to `nil` in the tests:

    stub_request(:get, 'http://username:password@localhost:8080/api/v1/pods')
      .to_raise(KubeException.new(401, error_message, nil))

(and I wonder how come the test are still working, we should check what's the logic in `to_raise` and what's comparing)